src/main/java/com/dabsquared/gitlabjenkins/trigger/handler/AbstractWebHookTriggerHandler.java

```diff
@@ -62,22 +62,26 @@ public void handle(Job<?, ?> job, H hook, boolean ciSkip, BranchFilter branchFil
     protected abstract boolean isCiSkip(H hook);
 
     private void setCommitStatusPendingIfNecessary(Job<?, ?> job, H hook) {
-        String buildName = PendingBuildsHandler.resolvePendingBuildName(job);
-        if (StringUtils.isNotBlank(buildName)) {
-            GitLabClient client = job.getProperty(GitLabConnectionProperty.class).getClient();
-            BuildStatusUpdate buildStatusUpdate = retrieveBuildStatusUpdate(hook);
-            try {
-                if (client == null) {
-                    LOGGER.log(Level.SEVERE, "No GitLab connection configured");
-                } else {
-                    String ref = StringUtils.removeStart(buildStatusUpdate.getRef(), "refs/tags/");
-                    String targetUrl = DisplayURLProvider.get().getJobURL(job);
-                    client.changeBuildStatus(buildStatusUpdate.getProjectId(), buildStatusUpdate.getSha(),
-                        BuildState.pending, ref, buildName, targetUrl, BuildState.pending.name());
+        try {
+            String buildName = PendingBuildsHandler.resolvePendingBuildName(job);
+            if (StringUtils.isNotBlank(buildName)) {
+                GitLabClient client = job.getProperty(GitLabConnectionProperty.class).getClient();
+                BuildStatusUpdate buildStatusUpdate = retrieveBuildStatusUpdate(hook);
+                try {
+                    if (client == null) {
+                        LOGGER.log(Level.SEVERE, "No GitLab connection configured");
+                    } else {
+                        String ref = StringUtils.removeStart(buildStatusUpdate.getRef(), "refs/tags/");
+                        String targetUrl = DisplayURLProvider.get().getJobURL(job);
+                        client.changeBuildStatus(buildStatusUpdate.getProjectId(), buildStatusUpdate.getSha(),
+                                BuildState.pending, ref, buildName, targetUrl, BuildState.pending.name());
+                    }
+                } catch (WebApplicationException | ProcessingException e) {
+                    LOGGER.log(Level.SEVERE, "Failed to set build state to pending", e);
                 }
-            } catch (WebApplicationException | ProcessingException e) {
-                LOGGER.log(Level.SEVERE, "Failed to set build state to pending", e);
             }
+        } catch (NullPointerException e) {
+            e.getMessage();
```

Comment:
Could you use the logger to handle the exception output instead? With somethhing like
```Java
LOGGER.log(Level.WARNING, e.getMessage(), e);
```
like they do at https://github.com/jenkinsci/gitlab-plugin/blob/e9d54cc611e57133eb5869d5744e9121f78b6eec/src/main/java/com/dabsquared/gitlabjenkins/publisher/GitLabVotePublisher.java#L80? 

I will have to update the patch I have done with just `e.getMessage();`in a hurry later to be consistent.

Reply:
I had a feeling that i should Logger instead. Should have asked before making a PR. Thanks for the correction.